app/controllers/concerns/foreman/controller/provisioning_templates.rb

```
@@ -42,4 +43,21 @@ def process_template_kind
   def type_name_singular
     @type_name_singular ||= resource_class.to_s.underscore
   end
+
+  def global_registration_vars
+    permitted = Foreman::Plugin.all
+                               .map(&:allowed_registration_vars)
+                               .flatten.compact.uniq
+    organization = params['organization_id'].present? ? Organization.authorized(:view_organizations).find(params['organization_id']) : Organization.current
+    location = params['location_id'].present? ? Location.authorized(:view_locations).find(params['location_id']) : Location.current
```

It should raise an error on invalid input, not silently discard it. It's so painful to debug if it doesn't because for a long time it appears to work.